viewer/src/repository/cad/CachedRepository.ts

```diff
@@ -77,7 +83,21 @@ export class CachedRepository implements Repository {
         );
         return merge(
           cachedSectorObservable.pipe(flatMap(wantedSector => this._modelDataCache.get(this.cacheKey(wantedSector)))),
-          uncachedSectorObservable.pipe(this.loadSectorFromNetwork()),
+          uncachedSectorObservable.pipe(
+            flatMap(
+              async _ => this._semaphore.acquire(),
+              (wantedSector: WantedSector, ready: boolean) => {
+                return {
+                  wantedSector,
+                  ready
+                };
+              }
+            ),
+            filter((data: { wantedSector: WantedSector; ready: boolean }) => data.ready),
+            map((data: { wantedSector: WantedSector; ready: boolean }) => data.wantedSector),
+            this.loadSectorFromNetwork(),
+            tap(_ => this._semaphore.release())
```

Comment:
What happens when network request fails? Slot is lost until Semaphore is cleared? Would moving the aquire to within `loadSectorFromNetwork()` and wrap in a `try { aquire(); ...} finally { release(); }` be better/clearer?

Reply:
Good point. I will see if we can just do it all in one function instead. I found the filter-map-load-tap a bit hard to read myself.

Reply:
don't use finalize, there are some changes to be made if we are to use it (namely where we move the queue logic)
And I want to move it so it works on both the i3d f3d AND ctm files

Reply:
Added a catchError now. @j-bjorne you probably know best if this is used correctly.